### DISCUSSION TITLE
matching wildcard tokens that contain a period with dependency parser

### DISCUSSION THREAD

**Discussion by asquare** [explosion/spaCy#10888] [Help: Coding & Implementations] [answered]
Hi, I'm trying to match tokens that contain a period against a wildcard pattern using the DependencyMatcher. The (simplified) example below works if the period in the middle token is replaced with an underscore, but not with a period. Two periods however work fine. I need to be able to handle middle (`id`) tokens containing a period. 

Text (does not work) - 
```
get C9120.d244 traffic
```

Text (works) - 
```
get C9120.d244.x traffic
```

Pattern -
```
{'RIGHT_ID': 'get', 'RIGHT_ATTRS': {'POS': 'VERB'}}, 
{'RIGHT_ID': 'id', 'RIGHT_ATTRS': {}, 'LEFT_ID': 'get', 'REL_OP': '.'}, 
{'RIGHT_ID': 'status', 'RIGHT_ATTRS': {'POS': {'IN': ['NOUN', 'PROPN']}}, 'LEFT_ID': 'id', 'REL_OP': '.'}
```
Any suggestions? Thanks in advance!

**Comment by adrianeboyd** [comment, marked as answer]
Have you double-checked that both `C9120.d244` and `C9120.d244.x` are exactly one token with your tokenizer and that the other tokens have the expected POS?

As long as they're exactly one token, I don't think it's actually due to the `.` since `{}` should always match exactly one token, so it might be a different part of your pattern that's not matching like you expect.

With these kinds of short phrases, the tags are often incorrect because there's not enough context. In addition, if you're using one of the provided trained pipelines like `en_core_web_sm`, it may be because it's not the kind of data the tagger was trained on (mainly full newspaper-style sentences).

**Reply by asquare** [reply]
Yes, it works better with longer sentences. Thanks for your sage advice!
